src/compas/scene/sceneobject.py

```diff
@@ -59,6 +96,31 @@ def transformation(self):
     def transformation(self, transformation):
         self._transformation = transformation
 
+    @property
+    def transformation_world(self):
+        frame_stack = []
+        parent = self.parent
+        while parent:
+            if parent.frame:
+                frame_stack.append(parent.frame)
+            parent = parent.parent
+        matrices = [Transformation.from_frame(f) for f in frame_stack]
+        if matrices:
+            transformation_world = reduce(mul, matrices[::-1])
+        else:
+            transformation_world = Transformation()
+
+        if self.transformation:
+            transformation_world *= self.transformation
+
+        return transformation_world
```

Comment:
Added a `transformation_world` property. This is the calculated final transformation from the frame stack and local transformation.

Reply:
@gonzalocasas The world transformation is calculated here